src/fact_base.py

```diff
@@ -42,11 +46,9 @@ def __init__(self):
             signal.signal(signal.SIGINT, self.shutdown_listener)
             signal.signal(signal.SIGTERM, self.shutdown_listener)
 
-        self.args = program_setup(self.PROGRAM_NAME, self.PROGRAM_DESCRIPTION, self.COMPONENT)
-        self.work_load_stat = WorkLoadStatistic(component=self.COMPONENT)
-
     def shutdown_listener(self, signum, _):
-        logging.info(f'Received signal {signum}. Shutting down {self.PROGRAM_NAME}...')
+        if self._is_main_process():
+            logging.info(f'Received signal {signum}. Shutting down {self.PROGRAM_NAME}...')
         self.run = False
```

Comment:
If I'm not mistaken setting `self.run` for any other process than the main process does not do anything.

My suggestion is to change the listener to
```python
def shutdown_listener(self, signum, _):
    if not self._is_main_process():
        # Comment about why we ignore the signal, but still have the handler registered.
        return
    logging.info(f'Received signal {signum}. Shutting down {self.PROGRAM_NAME}...')
    self.run = False
```